src/helm/benchmark/contamination/compute_contamination_metrics.py

```diff
@@ -0,0 +1,248 @@
+    )
+    parser.add_argument(
+        "--output-ngrams",
+        type=str,
+        default=None,
+        help="Path to the file of overlapped ngrams. If not given, ngrams will not be output.",
+    )
+
+    args = parser.parse_args()
+
+    tags: Dict[str, str] = {}
+    if args.tags is not None:
+        try:
+            for tag_pair in args.tags.split(";"):
+                k, v = tag_pair.split("=")
+                tags[k] = v
+        except ValueError:
+            raise ValueError("Failed to parse the tags.")
+
+    tokenizer: LightTokenizer
+    if args.normalization == "none":
+        tokenizer = LightTokenizer()
+    elif args.normalization == "default":
+        tokenizer = DefaultTokenizer()
+    else:
+        raise ValueError(f"Normalization strategy {args.normalization} is not defined.")
+
+    input_file_paths: List[str]
+    if os.path.isdir(args.input_data):
+        input_file_paths = []
+        for file_path in glob.iglob(os.path.join(args.input_data, "**/*"), recursive=True):
+            if os.path.isfile(file_path):
+                input_file_paths.append(file_path)
+    else:
+        input_file_paths = [args.input_data]
+    print(f"The input data will be loaded from {input_file_paths}")
+
+    print(f"Loading scenario data from {args.scenario_data}")
+    scenarios = load_scenarios_from_jsonl(args.scenario_data)
+
+    # initialize the stats and ngram_index
+    all_contamination_stats: Dict[str, ContaminationStats]
+    ngram_index: DefaultDict[int, DefaultDict[Tuple[str], Set[tuple]]]
+    all_contamination_stats, ngram_index = create_stats_and_ngram_index(scenarios, N_VALUES, tokenizer)
+
+    # Initialize overlapped_ngrams
+    overlapped_ngrams: Optional[DefaultDict[int, DefaultDict[str, int]]]
+    if args.output_ngrams is None:
+        overlapped_ngrams = None
+    else:
+        overlapped_ngrams = defaultdict(lambda: defaultdict(int))
+
+    # commpute the stats
+    for input_file_index in range(len(input_file_paths)):
+        input_file_path: str = input_file_paths[input_file_index]
+        print(f"Computing contamination stats for file {input_file_index+1}/{len(input_file_paths)} {input_file_path}")
+        compute_scenario_file_contamination(
+            training_file_path=input_file_path,
+            n_values=N_VALUES,
+            file_format=args.input_format,
+            ngram_index=ngram_index,
+            tokenizer=tokenizer,
+            overlapped_ngrams=overlapped_ngrams,
+        )
+
+    stats_summaries: List[str] = []
+    for contamination_stats in all_contamination_stats.values():
+        stats_summaries.append(contamination_stats.generate_summary(tags))
+
+    with open(args.output_stats, "w") as f:
+        f.write("\n".join(stats_summaries))
+    print(f"Written {len(stats_summaries)} results to {args.output_stats}")
+
+    if args.output_ngrams is not None:
```

Comment:
Another way to avoid so many cases is always to output `output_ngrams`, but to set the maximum number of n-grams, so that you can always turn it off by setting it to 0, and in some cases we might want to have a smaller number to save memory

Reply:
Can the Beam implementation share the same code as the non-beam implementation, so we inherit all the functionality?